Finishing the axios setup with tests

Marcos-Peter opened the pull request.

Marcos-Peter committed:
Finishing the axios setup with tests

alanraso commented on src/index.ts:
```diff
@@ -6,15 +6,22 @@ import { typeDefs } from './type-defs/type-defs';
-AppDataSource.initialize()
-  .then(async () => console.log(`Connected to database.`))
-  .catch((error) => console.log(error));
+export async function runServer(isMain = false) {
```
Esse parâmetro me parece um pouco estranho pq ele dita se o servidor vai rodar ou não 🤔 
Nesse caso não seria mais simples e mais claro de entender se tiver duas funções, uma de setup do banco e  outra de setup do server? Pq assim se vc quiser fazer só o setup do banco, vc chama a só ela.

alanraso commented on src/index.ts:
```diff
@@ -6,15 +6,22 @@ import { typeDefs } from './type-defs/type-defs';
-  .then(async () => console.log(`Connected to database.`))
-  .catch((error) => console.log(error));
+export async function runServer(isMain = false) {
+  await AppDataSource.initialize().catch((error) => console.log(error));
```
Aqui ele não vai interromper a execução caso a conexão com o banco dê errado. Porém se ele não conectar com o banco praticamente nada vai funcionar. Será que não vale interromper mesmo?

alanraso commented on src/index.ts:
```diff
@@ -6,15 +6,22 @@ import { typeDefs } from './type-defs/type-defs';
+    server.listen({ port }).then(({ url }) => {
+      console.log(`[SERVER] - Server running at ${url}`);
+    });
+  }
```
Aqui tem um possível probleminha de programação assíncrona. Se por acaso vc for usar esse `runServer` como promise e quiser esperar finalizar a execução pra fazer algo, aqui como está sendo usado `then`, ele vai continuar a execução antes de terminar o `listen`.

Marcos-Peter committed:
[Code Review Fix] - index.ts

Removing some unnecessary parameters and fixing the server shutdown in case of database errors.

Marcos-Peter committed:
Merge branch 'develop' into feature/Install-and-setup-axios

Marcos-Peter commented:
Código refatorado com base nos comentários do CR

alanraso commented on src/index.ts:
```diff
@@ -6,32 +6,24 @@ import { typeDefs } from './type-defs/type-defs';
+  runDatabase();
 
-  if (isMain) {
-    startServer;
-  }
+  server.listen({ port });
```
Nenhuma dessas duas promises está com await, o que indica que ele vai iniciar essas tarefas emparalelo e vai finalizar a execução dessa função `runServer`. Então se vc for usar essa runServer e tiver que esperar terminar esse setup, vc n tem essa possibilidade. É isso msm que vc queria?
